src/main/java/service/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,23 @@
+package service;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final Map<String, Integer> fruitStorage;
```

Comment:
The field 'fruitStorage' should be private and not exposed directly. Encapsulation is one of the key principles in object-oriented programming. Consider providing access to this data through methods rather than directly exposing the field.